### PR TITLE
[SPARK-37181][PYTHON] pyspark.pandas.read_csv() should support latin-1 encoding

### PR DESCRIPTION
**Pull request by pralabhkumar**
### What changes were proposed in this pull request?
pyspark pandas will read csv with latin-1 encoding. 
```
from pyspark import pandas as ps
ps.read_csv("<>", encoding="latin-1")
```

### Why are the changes needed?
Earlier it throws "pyspark.sql.utils.illegalArgumentException: latin-1" whereas pandas read csv is able to read "latin-1" encoding. 
In order to have the same behavior of pd.read_csv and ps.read_csv this change is needed.

### Does this PR introduce any user-facing change?
Yes user will be able to provide latin-1 encoding with pyspark pandas

### How was this patch tested?
unit tests

### CODE REVIEW

**Comment by srowen** [comment]
Jenkins test this please

**Review by srowen — COMMENTED** [review]
This seems reasonable, as pandas supports this too (but weirdly, docs it like it only works for UTF encodings). I am not sure that this causes files to be read as ISO-8859-1 - the text based readers strongly assume UTF-8 in Spark (in Hadoop really). I haven't tested this recently though

File: python/pyspark/pandas/namespace.py [hunk 1]
```diff
@@ -328,6 +334,11 @@ def read_csv(
 
         reader.options(**options)
 
+        if encoding is not None:
+            if encoding in encoding_mapping:
+                encoding = encoding_mapping[encoding]
+            reader.option("encoding", encoding)
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/namespace.py:340]:
```suggestion
        if encoding is not None:
            reader.option("encoding", encoding_mapping.get(encoding, encoding))
```

**Comment by HyukjinKwon** [comment]
CSV supports encoding that is only ascii compatible due to newline parsing. It does support other encodings with `multiline` on. I think it's fine to go ahead.

**Review by HyukjinKwon — APPROVED** [review]

**Review by Yikun — APPROVED** [review]
LGTM, Looks like there are still many alias on encoding [1] to be support in pyspark, we could append mappings when we need them. 

[1] https://github.com/python/cpython/blob/main/Lib/encodings/aliases.py

**Comment by HyukjinKwon** [comment]
Oh, that;s a very good point @Yikun

**Comment by pralabhkumar** [comment]
@HyukjinKwon If you are ok with it , can we please merge it

**Comment by HyukjinKwon** [comment]
Okay, let's go ahead as is for now

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by pralabhkumar** [comment]
Thx @HyukjinKwon @Yikun . For reviewing

**Pull request closed without merging**